### PR TITLE
PWGPP-528 - integrate RootInteractive into recipe

### PR DESCRIPTION
**Pull request by miranov25**
Dear @dberzano and @ktf 

As we agreed during alice OFFLINE week I made RootInteractive  a pip package and I patched
alidist recipies to use that package.  **I would like to ask for review.**

### Modifications and test - see (long) JIRA PWGPP-528
* modification and pytests:
  * [Modification and pytest output]
* some integration test output
  * [JIRA TPC QA example from pytest]

[Modification and pytest output]: https://alice.its.cern.ch/jira/browse/PWGPP-528?focusedCommentId=231029&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-231029

[JIRA TPC QA example from pytest]:https://alice.its.cern.ch/jira/browse/PWGPP-528?focusedCommentId=231030&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-231030

### Integration with virtualenv. Questions and ask for help
Until now, I tested only the setup **without alidock.**
I would like to ask you about the implementation as I use virtual env.

* I would like to use RootInteractive/AliRoot/AliPhysics  sourcing from cvmfs
  * Ideally people should be  able to run and modify tutoraial exploring  data using JUPYER notebook
  * Write interactive dashboard using reciies from interactive tutorials

* Is it possible to integrate virtual environment officially into aliBuild, or do you prefer one alidock - one environment? 
  * In the ML project of my wife (https://www.embl.de/research/units/cbb/kreshuk/) they have virtual environment within alidock
  * My current implementation is "working" hack

Regards
Marian

### CODE REVIEW

**Comment by ktf** [comment]
Looks like you are missing some `numpy` dependency. Could you take a look?

**Comment by miranov25** [comment]
I did not check yet O2 package, only AliRoot/AliPhysics
What is the fastest way to reproduce error?
````
aliBuild init O2@dev --defaults o2
````

**Comment by ktf** [comment]
Just build Python-modules with `--defaults o2`.

File: python-modules.sh [hunk 1]
```diff
@@ -96,7 +97,7 @@ fi
 MATPLOTLIB_TAG="3.0.3"
 if [[ $ARCHITECTURE != slc* ]]; then
   # Simply get it via pip in most cases
-  env PYTHONUSERBASE=$INSTALLROOT pip3 install --user "matplotlib==$MATPLOTLIB_TAG"
```
Comment (ktf) [inline python-modules.sh]: Why did you remove this? I guess this is the culprit...

Comment (miranov25) [reply python-modules.sh]: Looks like it could  be problem ()
In my first builds (from virtualenv described in JIRA), I had to disable this --user flag.
Testing the code  again, reverting back --user, the 
````
aliBuild build Python-modules  --defaults o2 --debug
````
is working.
````
==> Build of Python-modules successfully completed on `miranov-Strix-17-GL703GE'.
    Your software installation is at:
      /home2/miranov/software/alicesw3/sw/ubuntu1804_x86-64
    You can use this package by loading the environment:
      alienv enter Python-modules/latest-o2
````
Now i'm waiting for full build

**Comment by miranov25** [comment]
@ktf
Could we start test again? Eventually discuss about the virtualenv integration.

**Comment by ktf** [comment]
Looks like this is a known issue with scikit-garden:

https://github.com/scikit-garden/scikit-garden/issues/23

**Comment by ktf** [comment]
I've added an attempted fix... Let's see how it goes...

**Comment by ktf** [comment]
Ok, with the extra pip install it seems to have worked. Could you confirm while I wait for the macOS tests to finish and then we merge?

**Comment by miranov25** [comment]
I see build/O2Suite/alidist_macOS failed but I assume this looks like  another problem not related to my change.
For me the last change was working. 
To be sure I started the build again
````aliBuild build AliPhysics --defaults root6 --debug```
I will let you know in ~ 30 minutes

While waiting I'm also experimenting with the Python documentation, using sphinx.
Do we have in ALICE some preference, resp some custom extensions?

**Comment by ktf** [comment]
Yes, it looks separate. @knopers8 can you have a look?

**Comment by miranov25** [comment]
I checked the build after modifications.
* Root, AliRoot were build properly
* pytest is working well
From my side we can go ahead

**Comment by knopers8** [comment]
@ktf This error is already fixed on the master branch. Do you need us to release a new version of QC?

**Comment by knopers8** [comment]
Well, the fix is actually present in v0.13.1, so a bump should be enough.

**Comment by ktf** [comment]
Could you please do the bump? Thanks.

**Comment by knopers8** [comment]
https://github.com/alisw/alidist/pull/1659
